spec/models/manageiq/providers/hawkular/middleware_manager_spec.rb

```diff
@@ -198,4 +198,101 @@ def timeline_domain_expectations(status)
       end
     end
   end
+
+  describe 'connectivity' do
+    let(:ems) { FactoryGirl.create(:ems_hawkular) }
+    let(:endpoint) { FactoryGirl.create(:endpoint) }
+    let(:config) do
+      {
+        :username => 'abcdefghij',
+        :password => '1234567890',
+        :host     => 'awesomehawkular.org',
+        :port     => 8080,
+        :tenant   => 'hawkular',
+      }
+    end
+    context 'connect' do
+      before do
+        allow(ems).to receive(:hostname).and_return(config[:host])
+        allow(ems).to receive(:port).and_return(config[:port])
+        allow(ems).to receive(:default_endpoint).and_return(endpoint)
+        allow(ems).to receive(:authentication_userid).and_return(config[:username])
+        allow(ems).to receive(:authentication_password).and_return(config[:password])
+      end
+
+      it 'non-ssl' do
+        expect(::Hawkular::Client).to receive(:new) do |params|
```

Comment:
What do you think on exposing connection status methods on the Hawkular gem? I generally dislike mocking methods, but exposing `connected?` and `connection_params` on the gem would make this test relevant.

We could start with it as a monkey patch and move it back to the gem as soon as we ship InventoryV4.

Reply:
If he can use Hawkular client to define circumstances under which the connection is provided, it would make the tests much usable. I like that idea.

Reply:
You have the connection parameters as attributes in the client, although in documentation comments they are declared as private:
https://github.com/hawkular/hawkular-client-ruby/blob/a4f5c64242b7f226dba20d25e8d938a49ccc5691/lib/hawkular/base_client.rb#L18-L19

As Cainã suggests, in this case, I think it's better to check the attributes of the returned connection object rather than mocking the `new` method. I guess we could just remove that yardoc private modifier in the client gem, because that data is quite important to be exposed.